src/socket.rs

```diff
@@ -15,10 +15,13 @@
 use std::io::Error;
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 
+#[cfg(all(target_os = "linux"))]
 use realm_io;
+#[cfg(all(target_os = "linux"))]
 use socket2::SockRef;
```

Comment:
maybe move import to something already covered by a `cfg` - you can import within a function

Reply:
nice